R/apa_print_metafor.R

```diff
@@ -0,0 +1,57 @@
+#' Format statistics (APA 6th edition)
+#'
+#' These methods take \code{rma} objects to create formatted character
+#' strings to report the results in accordance with APA manuscript guidelines.
+#'
+#'
+#' @return
+#'    \code{apa_print.metafor} returns a list containing the following components according to the input:
+#'
+#'    \describe{
+#'      \item{\code{statistic}}{A named list of character strings giving the test statistic (z), and
+#'      \emph{p} value for each term.}
+#'      \item{\code{estimate}}{A named list of character strings giving the descriptive estimates and confidence intervals
+#'          for each term.}
+#'      \item{\code{full_result}}{A named list of character strings comprised of \code{estimate} and \code{statistic} for each term.}
+#'      \item{\code{table$estimate}}{A data.frame containing the study label, estimate, standard error, test statistic for each observed study and the overall meta-analytic estimate with standard error, test statistic, and p value. This table can be passed to \code{\link{apa_table}}.}
+#'    }
+#'    #'      \item{\code{table$heterogeneity}}{A data.frame containing the number of observations, measure and method employed in the analysis as well as the I^2, Tau^2 Tau^2 standard error and results of Cochran test of heterogeneity. This table can be passed to \code{\link{apa_table}}.}
+#'    }
+#' @family apa_print
+#'
+apa_print.metafor <- function(
```

Comment:
Unless I'm missing something, for S3-method dispatch to work this functions needs to be named after the S3 class it is ment to operate on. In this case, I think, it should be either the specific `rma.uni` or more general `rma` depending on how general this method should be. Right?